aea/package_manager/base.py

```diff
@@ -145,9 +157,78 @@ def sync(
 
             sync_needed = True
             self._logger.info(f"{package_id} not found locally, downloading...")
-            package_id_with_hash = package_id.with_hash(self.packages[package_id])
+            package_id_with_hash = package_id.with_hash(packages[package_id])
             self.add_package(package_id=package_id_with_hash)
 
+        return sync_needed, hash_updates, package_updates
+
+    def sync(
+        self,
+        dev: bool = False,
+        third_party: bool = True,
+        update_packages: bool = False,
+        update_hashes: bool = False,
+    ) -> "PackageManager":
+        """Sync local packages to the remote registry."""
+
+        if update_packages and update_hashes:
+            raise ValueError(
+                "Both `update_packages` and `update_hashes` cannot be set to `True`."
+            )
+
+        self._logger.info(f"Performing sync @ {self.path}")
+
+        sync_needed = False
+
+        if third_party:
+            self._logger.info("Checking third party packages.")
+            (
+                _sync_needed,
+                hash_updates_third_party,
+                package_updates_third_party,
+            ) = self._sync(
+                packages=self.third_party_packages,
+                update_hashes=update_hashes,
+                update_packages=update_packages,
+            )
+            sync_needed = sync_needed or _sync_needed
+
+            if update_hashes and hash_updates_third_party:
+                third_party_package_id = "\n\t- ".join(
+                    map(str, hash_updates_third_party)
+                )
+                self._logger.warning(
+                    f"Hashes for follwing third party module has changed.\n\t- {third_party_package_id}"
+                )
+
+            if update_packages and package_updates_third_party:
+                self._logger.info("Updating third party packages.")
+                for package_id, _ in package_updates_third_party.items():
+                    self._logger.info(f"Updating {package_id}")
+                    self.update_package(package_id=package_id)
```

Comment:
if we update_hashes, we update a copy , which is returned as `hash_updates_third_party`. We log this but don't actually update anything.
https://github.com/valory-xyz/open-aea/blob/8826c0a7b42f5b466d36795c565d7cafc80cb86a/aea/package_manager/base.py#L129